src/main/java/org/broadinstitute/hellbender/tools/spark/pipelines/PileupSpark.java

```diff
@@ -0,0 +1,138 @@
+import org.broadinstitute.hellbender.utils.spark.LocusWalkerSpark;
+import scala.Tuple3;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+@CommandLineProgramProperties(summary = "Print read alignments in Pileup-style format on Spark",
+        oneLineSummary = "Print read alignments in Pileup-style format on Spark",
+        programGroup = SparkProgramGroup.class)
+public final class PileupSpark extends LocusWalkerSpark {
+    private static final long serialVersionUID = 1L;
+
+    private static final String VERBOSE_DELIMITER = "@"; // it's ugly to use "@" but it's literally the only usable character not allowed in read names
+
+    @Override
+    public boolean requiresReads() { return true; }
+
+    @Argument(doc="the output file path", shortName = StandardArgumentDefinitions.OUTPUT_SHORT_NAME, fullName = StandardArgumentDefinitions.OUTPUT_LONG_NAME, optional = false)
+    protected String outputFile;
+
+    /**
+     * In addition to the standard pileup output, adds 'verbose' output too. The verbose output contains the number of
+     * spanning deletions, and for each read in the pileup it has the read name, offset in the base string, read length,
+     * and read mapping quality.  These per read items are delimited with an '@' character.
+     */
+    @Argument(fullName = "showVerbose", shortName = "verbose", doc = "Add an extra verbose section to the pileup output", optional = true)
+    public boolean showVerbose = false;
+
+    /**
+     * This enables annotating the pileup to show overlaps with metadata from a Feature file(s). For example, if you provide a
+     * VCF and there is a SNP at a given location covered by the pileup, the pileup output at that position will be
+     * annotated with the corresponding source Feature identifier.
+     */
+    @Argument(fullName = "metadata", shortName = "metadata", doc = "Features file(s) containing metadata", optional = true)
+    public List<FeatureInput<Feature>> metadata = new ArrayList<>();
+
+    /**
+     * Adds the length of the insert each base comes from to the output pileup. Here, "insert" refers to the DNA insert
+     * produced during library generation before sequencing.
+     */
+    @Hidden
+    @Argument(fullName = "outputInsertLength", shortName = "outputInsertLength", doc = "Output insert length", optional = true)
+    public boolean outputInsertLength = false;
+
+    @Override
+    public ReadFilter makeReadFilter(){
+        return new WellformedReadFilter(getHeaderForReads())
+                .and(ReadFilterLibrary.MAPPED)
+                .and(ReadFilterLibrary.NOT_DUPLICATE)
+                .and(ReadFilterLibrary.PASSES_VENDOR_QUALITY_CHECK)
+                .and(ReadFilterLibrary.PRIMARY_ALIGNMENT);
+    }
+
+    @Override
+    protected void runTool(final JavaSparkContext ctx) {
+        getAlignments(ctx).map(pileupFunction(metadata, outputInsertLength, showVerbose)).saveAsTextFile(outputFile);
```

Comment:
This would indeed be nice, but it means that the function would capture the class instance variables in its closure, which means either that unnecessary data is transferred, or it would fail if there are objects that are not serializable (even if they are not used by the function).